### PR TITLE
Feature/sg improvements

### LINKED ISSUE #33: Do not use in-line security group rules
Terraform has two resources that can define security group rules: In-line inside `aws_security_group`, or as attachments with `aws_security_group_rule`. Terraform says [here](https://www.terraform.io/docs/providers/aws/r/security_group.html) that you should not use both in-line and attached rules:

> At this time you cannot use a Security Group with in-line rules in conjunction with any Security Group Rule resources. Doing so will cause a conflict of rule settings and will overwrite rules.

AWS [supports](https://docs.aws.amazon.com/efs/latest/ug/manage-fs-access-vpc-peering.html) mounting EFS file systems across VPC peering connections. Normally, once a VPC peering connection is established, you are able to reference security groups in the other VPC inside your security group rules. However, if that peering connection is inter-region (the VPC's are in different regions), then AWS [says](https://docs.aws.amazon.com/vpc/latest/peering/vpc-peering-security-groups.html) you cannot reference SG ID's:

> You cannot reference the security group of a peer VPC that's in a different region. Instead, use the CIDR block of the peer VPC.

So, when using the `cloudposse/terraform-aws-efs` module, the `security_groups` input cannot be used with inter-region VPC peering. This should be OK, as the module now outputs the created security group's ID, and so I can add a custom ingress rule later.

Except that doesn't work, since the module is using in-line rules inside `aws_security_group`, instead of attached rules with `aws_security_group_rule`.

Can you switch to using `aws_security_group_rule`, please?

### PR DESCRIPTION
**Pull request by TBeijen**
Fixes #33 

**ATT:** When moving from inline to attached sg rules, I ran into this error:

```
Error: [WARN] A duplicate Security Group rule was found on (sg-nnnnnnnnnnnnnnnn). This may be
a side effect of a now-fixed Terraform issue causing two security groups with
identical attributes but different source_security_group_ids to overwrite each
other in the state. See https://github.com/hashicorp/terraform/pull/2376 for more
information and instructions for recovery. Error message: the specified rule "peer: 0.0.0.0/0, ALL, ALLOW" already exists
```
The PR mentioned: https://github.com/hashicorp/terraform/pull/2376

It looks to be originating from the inline rules and the newly added rule attachment overlapping.

For this reason I renamed the `default` security group into `efs`, and suffixed the name as well.
This allows for a 0-downtime sg swap, where the new attachment-based sg replaces the old inline-based sg. Not sure if there's a better approach for this but seemed better than requiring a manual step that likely causes downtime.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -27,34 +27,40 @@ resource "aws_efs_mount_target" "default" {
...
+  tags = module.label.tags
+}
 
-  egress {
-    from_port   = 0
-    to_port     = 0
-    protocol    = "-1"
-    cidr_blocks = ["0.0.0.0/0"]
-  }
+resource "aws_security_group_rule" "inbound" {
```
Comment (aknysh) [inline main.tf]:
```suggestion
resource "aws_security_group_rule" "ingress" {
```

File: main.tf [hunk 2]
```diff
@@ -27,34 +27,40 @@ resource "aws_efs_mount_target" "default" {
...
+  type                     = "ingress"
+  from_port                = "2049" # NFS
+  to_port                  = "2049"
+  protocol                 = "tcp"
+  source_security_group_id = var.security_groups[count.index]
+  security_group_id        = aws_security_group.efs[0].id
+}
 
-  tags = module.label.tags
+resource "aws_security_group_rule" "outbound" {
```
Comment (aknysh) [inline main.tf]:
```suggestion
resource "aws_security_group_rule" "egress" {
```

File: main.tf [hunk 3]
```diff
@@ -27,34 +27,40 @@ resource "aws_efs_mount_target" "default" {
...
-    cidr_blocks = ["0.0.0.0/0"]
-  }
+resource "aws_security_group_rule" "inbound" {
+  count                    = var.enabled ? length(var.security_groups) : 0
+  type                     = "ingress"
+  from_port                = "2049" # NFS
+  to_port                  = "2049"
+  protocol                 = "tcp"
+  source_security_group_id = var.security_groups[count.index]
+  security_group_id        = aws_security_group.efs[0].id
```
Comment (aknysh) [inline main.tf]:
```suggestion
  security_group_id        = join("", aws_security_group.efs.*.id)
```

File: main.tf [hunk 4]
```diff
@@ -27,34 +27,40 @@ resource "aws_efs_mount_target" "default" {
...
 
-  tags = module.label.tags
+resource "aws_security_group_rule" "outbound" {
+  count             = var.enabled ? 1 : 0
+  type              = "egress"
+  from_port         = 0
+  to_port           = 0
+  protocol          = "-1"
+  cidr_blocks       = ["0.0.0.0/0"]
+  security_group_id = aws_security_group.efs[0].id
```
Comment (aknysh) [inline main.tf]:
```suggestion
  security_group_id = join("", aws_security_group.efs.*.id)
```

**Comment by TBeijen** [comment]
@aknysh Done

**Pull request merged**